### DISCUSSION TITLE
How to import/export custom functions

### DISCUSSION THREAD

**Discussion by isNotOkay** [pawelsalawa/sqlitestudio#4797] [General]
Hello, I would like to import/export custom javascript functions that I created with the function editor. Is this possible?

**Comment by pawelsalawa** [comment]
There is no designated mechanism to export/import these, but you can just copy a configuration file to another computer, if you want to move your functions there. The file location is described at https://github.com/pawelsalawa/sqlitestudio/wiki/Configuration_directory_location

**Reply by isNotOkay** [reply]
Thanks!

**Reply by ve3meo** [reply]
Where in the settings db are functions stored?

I find nothing in the settings table.

**Reply by pawelsalawa** [reply]
`select * from settings where [group] = 'Internal' and [key] = 'Functions';`
Note, that its value is stored in binary form (functions are serialized from `QHash` object), so you won't be able just view it as text.

**Reply by ve3meo** [reply]
Thanks. I looked there, even CAST [value] to text, but SQLiteSpy revealed nothing. However SQLiteStudio's Form view does. Would it follow that migrating that one record to the settings table for SQLite Studio on another device would transfer the functions from one to another intact?

**Reply by pawelsalawa** [reply]
I haven't tested it, but in theory it should work.
